### PR TITLE
Update: comma-dangle.md

### PR DESCRIPTION
**Pull request by CurtisHumphrey**
**What is the purpose of this pull request? (put an "X" next to item)**

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

**Is there anything you'd like reviewers to focus on?**

I think this addition is needed as without it one, like me, would have to go to source code to figure out why it doesn't work.

### CODE REVIEW

File: docs/rules/comma-dangle.md [hunk 1]
```diff
@@ -60,6 +60,17 @@ This rule has a string option or an object option:
 * `"always-multiline"` requires trailing commas when the last element or property is in a *different* line than the closing `]` or `}` and disallows trailing commas when the last element or property is on the *same* line as the closing `]` or `}`
 * `"only-multiline"` allows (but does not require) trailing commas when the last element or property is in a *different* line than the closing `]` or `}` and disallows trailing commas when the last element or property is on the *same* line as the closing `]` or `}`
 
+Trailing commas when the last element is a spread are not allowed even when always is on.
```
Comment (vitorbal) [inline docs/rules/comma-dangle.md]: I think we should change the wording here a bit. The spec forbids trailing commas on AssignmentRestElement nodes, it's not that ESLint arbitrarily decided to not allow them for this rule in specific (see [this issue](https://github.com/eslint/eslint/issues/3627) for some more context). I would suggest something like "Please note that trailing commas are not allowed by the spec".

Comment (vitorbal) [inline docs/rules/comma-dangle.md]: I think this paragraph should come before the "Options" sub-section, inside the "Rule details" sub-section.

File: docs/rules/comma-dangle.md [hunk 2]
```diff
@@ -60,6 +60,17 @@ This rule has a string option or an object option:
...
+
+Always ***incorrect***
+
+```js
+const {
+ first,
+ ...rest,
+} = array;
```
Comment (vitorbal) [inline docs/rules/comma-dangle.md]: `array` is misleading, this should be an object. How about `obj`?

File: docs/rules/comma-dangle.md (hunk 1 above)
Comment (not-an-aardvark) [inline docs/rules/comma-dangle.md]: Also, I think you're referring to rest elements, not spread elements.

Comment (vitorbal) [inline docs/rules/comma-dangle.md]: @not-an-aardvark indeed, good catch.

**Comment by vitorbal** [comment]
@CurtisHumphrey just checking in, are you still planning on finishing this up?

**Comment by kaicataldo** [comment]
@CurtisHumphrey One more friendly ping.

**Comment by gyandeeps** [comment]
Closing because of no response. @CurtisHumphrey Feel free to reopen.

**Pull request closed without merging**